test/libdl.jl

```diff
@@ -171,7 +171,7 @@ let dl = C_NULL
     end
 end
 
-if OS_NAME in (:Linux, :FreeBSD)
+if Sys.KERNEL in (:Linux, :FreeBSD)
```

Comment:
the function is defined based on compiler macros which aren't guaranteed to match the uname that this variable comes from. check via dlsym maybe?